packages/design-system/src/components/Button/Button.tsx

```diff
@@ -1,7 +1,43 @@
+import {
+  AnalyticsEventProps,
+  EVENT_CATEGORY,
+  sendAnalyticsEvent,
+} from '../analytics/SendAnalytics';
 import React from 'react';
+import { buttonSendsAnalytics } from '../flags';
 import classNames from 'classnames';
 
+export type ButtonComponent = React.ReactElement<any> | any | ((...args: any[]) => any);
+export type ButtonSize = 'small' | 'big';
+export type ButtonType = 'button' | 'submit';
+/**
+ * A string corresponding to the button-component variation classes.
+ * The danger variation is deprecated and will be removed in a future release.
+ */
+export type ButtonVariation = 'primary' | 'danger' | 'success' | 'transparent';
```

Comment:
note to self: need to add `'secondary' | 'tertiary'` to this list once [PR](https://github.com/CMSgov/design-system/pull/1106/files) is merged